state/state.go

```diff
@@ -34,6 +40,110 @@ type DownloadedObject struct {
 	Object, Filename string
 }
 
+type DynamoDBStater struct {
+	*sync.Mutex
+	Session *session.Session
+	Service string
+}
+
+func NewDynamoDBStater(session *session.Session, service string) *DynamoDBStater {
+	return &DynamoDBStater{
+		Mutex:   &sync.Mutex{},
+		Session: session,
+		Service: service,
+	}
+}
+
+// list of processed objects
+func (d *DynamoDBStater) processedObjects() (map[string]time.Time, error) {
+	objs := make(map[string]time.Time)
+
+	type Record struct {
```

Comment:
Let's define this type outside of this function, since it's used in both processing listed objects and setting newly completed objects in `SetProcessed`.

Reply:
+1

Also:

1. Nitpick: preference for `S3Object` over `S3object` (Go casing conventions)
2. `json:"S3object"` is redundant - the JSON tags are only needed if the name for the key they will be marshalled to will be different, e.g. `s3_object`. So you can probably leave the `json:""` tags out if you're good with those key names in the final table.

Reply:
Although, re: `Record`, is it possible to use `map[string]time.Time` everywhere instead of having separate types that marshal up? https://github.com/honeycombio/honeyaws/blob/master/state/state.go#L58